Fix display of ParseFile in Config

natanrolnik opened the pull request:
This PR fixes the display of `Parse.File`s  and GeoPoints in the Config section of the Dashboard.
Now that @flovilmart fixed saving Files to Config in Parse Server (ParsePlatform/parse-server#3457), this is the last bit to have it working in the dashboard.

### Before:
![captura de tela 2017-02-20 as 13 30 17](https://cloud.githubusercontent.com/assets/1164565/23123401/d9c50bd2-f770-11e6-916a-ef7be1241e59.png)

![captura de tela 2017-02-20 as 13 19 02](https://cloud.githubusercontent.com/assets/1164565/23123187/19612704-f770-11e6-9c28-44aeca3c97ec.png)

### After
<img width="973" alt="captura de tela 2017-02-19 as 23 51 24" src="https://cloud.githubusercontent.com/assets/1164565/23107619/04140356-f709-11e6-8df6-e4345bff6163.png">

![captura de tela 2017-02-20 as 13 19 08](https://cloud.githubusercontent.com/assets/1164565/23123204/26053798-f770-11e6-99b3-17f3d7b2e116.png)


@JeremyPlease @dvanwinkle @steven-supersolid if anyone of you could review, it would be great.

flovilmart commented on src/dashboard/Data/Config/Config.react.js:
```diff
@@ -72,6 +72,8 @@ class Config extends TableView {
     let value = data.value;
     let modalValue = value;
     let type = typeof value;
+    let openModalWhenClinkingValue = true;
```
remove that and handle in openModal

flovilmart commented on src/dashboard/Data/Config/Config.react.js:
```diff
@@ -83,10 +85,11 @@ class Config extends TableView {
       } else if (value instanceof Parse.File) {
         type = 'File';
         value = <a target='_blank' href={value.url()}>Open in new window</a>;
+        openModalWhenClinkingValue = false;
```
remove that :)

flovilmart commented on src/dashboard/Data/Config/Config.react.js:
```diff
@@ -105,11 +108,19 @@ class Config extends TableView {
       <tr key={data.param}>
         <td style={columnStyle} onClick={openModal}>{data.param}</td>
         <td style={columnStyle} onClick={openModal}>{type}</td>
-        <td style={columnStyle} onClick={openModal}>{value}</td>
```
revert that :)

flovilmart commented on src/dashboard/Data/Config/Config.react.js:
```diff
@@ -105,11 +108,19 @@ class Config extends TableView {
     });
     let columnStyle = { width: '30%', cursor: 'pointer' };
+
+    let valueColumn;
```
remove all that, and check in `openModal`

```
let openModal = () => this.setState({
       modalOpen: true,
       modalParam: data.param,
       modalType: type,
        modalValue: modalValue
      });

// to:

let openModal = () => {
  if (modalValue instanceof Parse.File) { return }
  this.setState({
       modalOpen: true,
       modalParam: data.param,
       modalType: type,
       modalValue: modalValue
  });
}
```

natanrolnik commented on src/components/FileInput/FileInput.react.js:
```diff
@@ -40,7 +40,7 @@ export default class FileInput extends React.Component {
   render() {
     let inputProps = {
       type: 'file',
-      value: null,
```
This was generating a warning with files

natanrolnik commented:
@flovilmart thanks for your review! 
I indeed could do it in a more elegant way, and I updated. But I couldn't move the `if (modalValue instanceof Parse.File) { return }` condition to the `openModal` callback. This is because I **do** want to open the modal when it's a file, I just don't want it to be triggered from the last column. So I kinda created a mix of our two solutions, making it cleaner.
As a plus, I've also fixed the display of GeoPoints.
